src/main/java/com/heatheryou/httpserver/RequestParser.java

```diff
@@ -9,18 +10,33 @@
 
 public class RequestParser {
     private List<String> requestList;
-    String uri;
-    String method;
-    String body;
+    private ISystemOutput systemOutput;
 
-    public Request processRequest(BufferedReader requestReader) throws IOException {
+    public RequestParser(ISystemOutput systemOutput) {
+        this.systemOutput = systemOutput;
+    }
+
+    public Request processRequest(BufferedReader requestReader) {
+        try {
+            return routeValidRequest(requestReader);
+        } catch (Exception e) {
+            return routeInvalidRequest();
+        }
+    }
+
+    private Request routeValidRequest(BufferedReader requestReader) throws IOException {
         List<String> requestList = readRequest(requestReader);
         String[] requestLine = parse(requestList);
         int contentLength = getContentLength(requestList);
-        setRequest(requestReader, requestLine, contentLength);
-        return getRequest();
+        return getRequest(requestReader, requestLine, contentLength);
+    }
+
+    private Request routeInvalidRequest() {
+        systemOutput.printErr("Exception: Bad Request");
+        return new Request(null, null, null);
```

Comment:
Having to return a `Request` with all `null`s like this makes me think that the design could be improved somewhere.  I don't have a solid suggestion for how at the moment.

Reply:
Yes, I agree. I originally intended to have this Request route to a 400 somehow, and the compiler is not happy unless I `return` something, so this is what I have for now.